src/gui/mesh/qgsmeshrendererscalarsettingswidget.cpp

```diff
@@ -44,10 +49,10 @@ QgsMeshRendererScalarSettingsWidget::QgsMeshRendererScalarSettingsWidget( QWidge
 
   // connect
   connect( mScalarRecalculateMinMaxButton, &QPushButton::clicked, this, &QgsMeshRendererScalarSettingsWidget::recalculateMinMaxButtonClicked );
-  connect( mScalarMinLineEdit, &QLineEdit::textChanged, this, &QgsMeshRendererScalarSettingsWidget::minMaxChanged );
-  connect( mScalarMaxLineEdit, &QLineEdit::textChanged, this, &QgsMeshRendererScalarSettingsWidget::minMaxChanged );
-  connect( mScalarMinLineEdit, &QLineEdit::textEdited, this, &QgsMeshRendererScalarSettingsWidget::minMaxEdited );
-  connect( mScalarMaxLineEdit, &QLineEdit::textEdited, this, &QgsMeshRendererScalarSettingsWidget::minMaxEdited );
+  connect( mScalarMinSpinBox, qOverload<double>( &QgsDoubleSpinBox::valueChanged ), this, [ = ]( double ) { minMaxChanged(); } );
+  connect( mScalarMaxSpinBox, qOverload<double>( &QgsDoubleSpinBox::valueChanged ), this, [ = ]( double ) { minMaxChanged(); } );
+  connect( mScalarMinSpinBox, &QgsDoubleSpinBox::editingFinished, this, &QgsMeshRendererScalarSettingsWidget::minMaxEdited );
```

Comment:
this breaks the update of the classification when user change min/max

Reply:
Before, when user change the value, even if he doesn't finished to enter the value, the classification was updated. I just see that `minMaxEdited` was always called.
I wonder was it is the best :  
* the classification is updated as before (if user type "12", first update for "1" then for "2")
* or wait for the spin box lost the focus or the user press enter.
I don't have a strong opinion on this.
For both case, it seems that  `minMaxChanged` does not serve a lot... Further more, it breaks your implementation: first update min/max without updating classification, then when `minMaxEdited` is called, as the min/max are the same, the classification is not updated.
I would just use `minMaxEdited` for both signals (or only for `editingFinished` if we don't  want update of the classification when editing)

Reply:
How about this: https://github.com/qgis/QGIS/pull/43831/commits/1600e25b47f6b30a46959e97e2ee7a9eed493e99 ?

Reply:
I tried it and it seems ok, thanks !

Reply:
@vcloarec thank you for the review!